app/src/androidTestE2eTest/java/io/appium/uiautomator2/unittest/test/internal/TestUtils.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.appium.uiautomator2.unittest.test.internal;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.uiautomator.UiDevice;
+import android.support.test.uiautomator.Until;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+
+import io.appium.uiautomator2.model.By;
+import io.appium.uiautomator2.unittest.test.Config;
+import io.appium.uiautomator2.utils.Device;
+
+import static android.os.SystemClock.elapsedRealtime;
+import static io.appium.uiautomator2.unittest.test.Config.APP_LAUNCH_TIMEOUT;
+import static io.appium.uiautomator2.unittest.test.Config.DEFAULT_POLLING_INTERVAL;
+import static io.appium.uiautomator2.unittest.test.Config.IMPLICIT_TIMEOUT;
+import static io.appium.uiautomator2.unittest.test.internal.commands.DeviceCommands.findElement;
+import static io.appium.uiautomator2.unittest.test.internal.commands.DeviceCommands.source;
+import static io.appium.uiautomator2.unittest.test.internal.commands.ElementCommands.click;
+import static io.appium.uiautomator2.unittest.test.internal.commands.ElementCommands.getName;
+import static io.appium.uiautomator2.utils.Device.getUiDevice;
+
+@SuppressWarnings("JavaDoc")
+public class TestUtils {
+
+    private static final int APP_LAUNCH_RETRIES_COUNT = 3;
+    private static final String PM_GRANT_COMMAND = "pm grant %s %s";
+
+    public static void waitForSeconds(int seconds) {
+        try {
+            Thread.sleep(seconds * 1000);
+        } catch (InterruptedException ignore) {
+        }
+    }
+
+    public static void waitForMillis(int millis) {
+        try {
+            Thread.sleep(millis);
```

Comment:
I assume in Android world it's more common to use SystemClock class for this purpose

Reply:
https://developer.android.com/reference/android/os/SystemClock.html
>SystemClock.sleep(millis) is a utility function very similar to Thread.sleep(millis), but it ignores InterruptedException.